lib/expression/v2.ex

```diff
@@ -82,12 +82,17 @@ defmodule Expression.V2 do
   end
 
   @doc """
-  This function is referenced by `Expression.V2.Compile` to 
+  This function is referenced by `Expression.V2.Compile` to
   make access to values in Maps or Lists easier
   """
   @spec read_attribute(map | list, binary | integer) :: term
   def read_attribute(map, item) when is_map(map), do: Map.get(map, item)
-  def read_attribute(list, index) when is_list(list), do: Enum.at(list, index)
+
+  def read_attribute(list, index) when is_list(list) and is_integer(index),
+    do: Enum.at(list, index)
+
+  def read_attribute(list, range) when is_list(list) and is_struct(range, Range),
+    do: Enum.slice(list, range)
```

Comment:
`foo[0...3]` is a syntax that's supported in V1, needed to introduce it in to V2.